plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,440 @@
+	if diskSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_DiskSpaceMonitor; diskSpaceMonitor != nil {
+		diskSpaceMonitorRoute := "hudson.node_monitors.DiskSpaceMonitor"
+		var diskSpace map[string]interface{}
+		if diskSpace, ok = diskSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, diskSpaceMonitor, "map[string]interface{}", diskSpaceMonitorRoute)
+		}
+		if tags["disk_path"], ok = diskSpace["path"].(string); !ok {
+			return badFormatErr(url, diskSpace["path"], "string", diskSpaceMonitorRoute+".path")
+		}
+		if fields["disk_available"], ok = diskSpace["size"].(float64); !ok {
+			return badFormatErr(url, diskSpace["size"], "float64", diskSpaceMonitorRoute+".size")
+		}
+	}
+
+	if tempSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor; tempSpaceMonitor != nil {
+		tempSpaceMonitorRoute := "hudson.node_monitors.TemporarySpaceMonitor"
+		var tempSpace map[string]interface{}
+		if tempSpace, ok = tempSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, tempSpaceMonitor, "map[string]interface{}", tempSpaceMonitorRoute)
+		}
+		if tags["temp_path"], ok = tempSpace["path"].(string); !ok {
+			return badFormatErr(url, tempSpace["path"], "string", tempSpaceMonitorRoute+".path")
+		}
+		if fields["temp_available"], ok = tempSpace["size"].(float64); !ok {
+			return badFormatErr(url, tempSpace["size"], "float64", tempSpaceMonitorRoute+".size")
+		}
+	}
+
+	if swapSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor; swapSpaceMonitor != nil {
+		swapSpaceMonitorRouter := "hudson.node_monitors.SwapSpaceMonitor"
+		var swapSpace map[string]interface{}
+		if swapSpace, ok = swapSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, swapSpaceMonitor, "map[string]interface{}", swapSpaceMonitorRouter)
+		}
+		if fields["swap_available"], ok = swapSpace["availableSwapSpace"].(float64); !ok {
+			return badFormatErr(url, swapSpace["availableSwapSpace"], "float64", swapSpaceMonitorRouter+".availableSwapSpace")
+		}
+		if fields["swap_total"], ok = swapSpace["totalSwapSpace"].(float64); !ok {
+			return badFormatErr(url, swapSpace["totalSwapSpace"], "float64", swapSpaceMonitorRouter+".totalSwapSpace")
+		}
+		if fields["memory_available"], ok = swapSpace["availablePhysicalMemory"].(float64); !ok {
+			return badFormatErr(url, swapSpace["availablePhysicalMemory"], "float64", swapSpaceMonitorRouter+".availablePhysicalMemory")
+		}
+		if fields["memory_total"], ok = swapSpace["totalPhysicalMemory"].(float64); !ok {
+			return badFormatErr(url, swapSpace["totalPhysicalMemory"], "float64", swapSpaceMonitorRouter+".totalPhysicalMemory")
+		}
+	}
+	acc.AddFields(measurementNode, fields, tags)
+
+	return nil
+}
+
+func (j *Jenkins) gatherNodesData(acc telegraf.Accumulator) {
+	var nodes []*gojenkins.Node
+	var err error
+	err = j.doGet(func() error {
+		nodes, err = j.instance.GetAllNodes()
+		return err
+	})
+
+	url := j.URL + "/computer/api/json"
```

Comment:
Throughout the pull request, don't build URLs for the error strings.  If an error occurs while reading a node, just return the name of the node.  If an error occurs with a build, return the jobname and build_id.

Reply:
👍